### PR TITLE
Public Schedule: Clicking on a filter moves the page #5826

### LINKED ISSUE #5826: Public Schedule: Clicking on a filter moves the page
Clicking on any filter on the public schedule move the page up. This is contrary to how this works on other services. Filtering items should simply apply the filter, but it should not move the page. Compare https://eventyay.com/e/ae73f280/schedule

Screenshot from 2020-11-29 21-56-27

### PR DESCRIPTION
**Pull request by akash2408**
Fixes #5826

#### Short description of what this resolves:
adding ember install ember-router-scroll

#### Changes proposed in this pull request:
-
-
-

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by sachinchauhan2889 — CHANGES_REQUESTED** [review]
issue is still there. page is still moving after clicking filtering sessions.

File: app/controllers/public/sessions.js [hunk 1]
```diff
@@ -20,11 +19,7 @@ export default class SessionsController extends Controller {
 
   @action
   sessionChanged() {
-    if (this.model.isFiltering) {
-      $('html,body').animate({
-        scrollTop: $('#session-heading').offset()?.top
-      });
-    }
+    this.set('preserveScrollPosition', true);
```
Comment (sachinchauhan2889) [inline app/controllers/public/sessions.js:22]: why created this.

**Review by sachinchauhan2889 — CHANGES_REQUESTED** [review]
Try
just remove sessionChanged() action----> line 21-29. 
and also remove 'sessionChanged' from this.addobserver()----> in line 18.

according to me this resolve issue.

File: app/controllers/public/sessions.js (hunk 1 above)
Comment (akash2408) [reply app/controllers/public/sessions.js:22]: After only removing session changed whenever we click on filter page will reload and moves up using previous scoll position this well save the earlier scroll postion of page and didnt moves the page up and down

Comment (sachinchauhan2889) [reply app/controllers/public/sessions.js:22]: you are right. :+1:

**Review by sachinchauhan2889 — APPROVED** [review]

**Comment by sachinchauhan2889** [comment]
@iamareebjamal sir, look good to me.

please review

File: app/controllers/public/sessions.js (hunk 1 above)
Comment (sachinchauhan2889) [reply app/controllers/public/sessions.js:22]: good work @akash2408

**Comment by mariobehling** [comment]
Is not working. The page should not move at all. Date, track and rooms still result in moving the page. The page should not move at all where-ever it is. It currently move to the heading session.

**Comment by akash2408** [comment]
> It is not working. The page should not move at all. Date, track and rooms still result in moving the page. The page should not move at all where-ever it is. It currently moves to the heading session.

Sir, Thats why i remove the content of sesssion changed function.The gif was of previous commit .please review again

**Comment by iamareebjamal** [comment]
@akash2408 That's because you pushed from development branch. Never ever create a PR from development branch. Now, we can't preview your work

**Comment by iamareebjamal** [comment]
Superseded by #5878

**Pull request closed without merging**